.github/workflows/rootski-ci.yml

```diff
@@ -34,6 +34,21 @@ jobs:
       with:
         python-version: "3.9"
 
+    # cache the python environment for faster dependency installation
+    # it's important to use env.pythonLocation to get the current python version
+    - name: cache root environment
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.pythonLocation }}
+        key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}
```

Comment:
Based on the file hash (the hash changes when the file is modified) these actions will store a new key if it doesn't find a match and update the cached path. If the hash matches it restores the cache to the directory.

@phitoduck I'm not sure if we want to do this now, but we should list any files on these steps that could change the dependencies downloaded. Since it's a mono repo, there might be a few to list here.